pyvista/core/datasetattributes.py

```diff
@@ -30,20 +44,83 @@ class DataSetAttributes(_vtk.VTKObjectWrapper):
 
     association : FieldAssociation
         The array association type of the vtkobject.
+
+    Examples
+    --------
+    Store data with point association in a DataSet
+
+    >>> import pyvista
+    >>> mesh = pyvista.Cube().clean()
+    >>> mesh.point_arrays['my_data'] = range(mesh.n_points)
+    >>> data = mesh.point_arrays['my_data']
+    >>> data
+    pyvista_ndarray([0, 1, 2, 3, 4, 5, 6, 7])
+
+    Change the data array and show that this is reflected in the DataSet.
+
+    >>> data[:] = 0
+    >>> mesh.point_arrays['my_data']
+    pyvista_ndarray([0, 0, 0, 0, 0, 0, 0, 0])
+
     """
 
-    def __init__(self, vtkobject: _vtk.vtkFieldData, dataset: _vtk.vtkDataSet, association: FieldAssociation):
+    def __init__(self, vtkobject: _vtk.vtkFieldData, dataset: _vtk.vtkDataSet,
+                 association: FieldAssociation):
         """Initialize DataSetAttributes."""
         super().__init__(vtkobject=vtkobject)
         self.dataset = dataset
         self.association = association
 
     def __repr__(self) -> str:
         """Printable representation of DataSetAttributes."""
+        array_info = ' None'
+        if len(self):
+            lines = []
+            for name, array in self.items():
+                if len(name) > 20:
+                    name = f'{name[:20]}...'
+                lines.append(f'{name[:23]:<23} {str(array.dtype):<10} {array.shape}')
```

Comment:
Can we also add what kind of data it is here, scalar vs. vector?  I can't figure out how to inspect whether a data array is a scalar or a vector.  Maybe I'm missing it.

Reply:
```py
import pyvista
cube = pyvista.Cube()
cube.point_arrays.GetArray("Normals").GetArrayType()
```

yields `2`, which is `NORMALS`.  This is good!  But, we don't really handle this natively as an array type as far as I can tell in this PR.  Or is this an implicit usage that we don't need explicit methods for?

Reply:
scratch that, `GetArrayType` is not the right method.  It is

```python
import pyvista
cube = pyvista.Cube()
cube.point_arrays.IsArrayAnAttribute(0)  # Normals?  -> 1
cube.point_arrays.IsArrayAnAttribute(1)  # TCoords?  -> 3
```

So Normals is stored as a VECTOR in Cube while TCoords is stored as TCOORDS.

Reply:
Thanks for pointing this out.  I noticed this wasn't being done right in glTF, but didn't quite connect the dots that NORMALS needed to be an attribute.  This seems to be a good PR to add this to.

Reply:
I'm getting this correctly reported:

```py
>>> import pyvista
>>> mesh = pyvista.Cube().clean()
>>> mesh.clear_arrays()
>>> mesh = mesh.compute_normals(cell_normals=False)
>> dset = mesh.point_arrays.VTKObject
>>> dset.IsArrayAnAttribute(0)
2
```

Regardless, I need to add normals as an attribute.